### ISSUE TITLE
Usability: Full screen editor makes side and top bar hard to access

### ISSUE THREAD

**Issue by wmalexander** [WordPress/gutenberg#22178]
The default full screen mode hides so much navigation behind extra clicks as to make it less intuitive. The interaction for "new post" as well as "view post" each are difficult to find, for both long time and new users alike, and could be displayed on the edit screen without the need for extra clicks. [Please see this video demonstration](https://www.loom.com/share/6ae9a43d049b43e7a788129dff66bfec). 

Having been a WordPress user for over a decade, it took me 10 minutes of discussion with a colleague to determine that I could make these interactions more accessible by exiting fullscreen mode - and even then, fullscreen toggling requires extra clicks to find. In the pre-gutenberg editor, viewing a post, or adding a new post from the edit screen could be accomplished in one click. In the current version this requires a minimum of two clicks, and depending on the approach used, can require navigating to the dashboard from the edit page, requiring a page load. This makes the user experience less intuitive and more challenging, especially for users new to WordPress or to the new editor, but for long-time users as well. 

- WordPress version: 5.4.1
- Default installation

All OS, Browsers, Devices etc.

**Comment by talldan** [comment]
There is work on https://github.com/WordPress/gutenberg/pull/22191 to make it easier to access the admin sidebar. I note that this PR doesn't have a linked issue, so lets use this one to track things 👍.

There's also a separate issue about also accessing the top admin bar - https://github.com/WordPress/gutenberg/issues/21739.

**Comment by dingo-d** [comment]
The #22191 doesn't really address this issue. What is needed is a quick and easy way to view the page at any time. Not just on update. An update will throw notification, that will disappear after a few moments. 

It would be great to have view post (page) link somewhere visible and accessible at all times...

**Comment by wmalexander** [comment]
Yes what I'm suggesting is a "View Page" button and "New Page" link such as in the top right corner of this screenshot:
<img width="1084" alt="Screen Shot 2020-05-12 at 6 58 12 AM" src="https://user-images.githubusercontent.com/16945224/81676635-20e48b80-941e-11ea-8bf1-e20ace846c1e.png">

**Comment by hansschuijff** [comment]
Hi All,

I agree totally with the problem analysis of William @wmalexander and had written a support question about [the ommission of clear exits from the  full screen editor mode](https://wordpress.org/support/topic/no-obvious-way-to-exit-the-editor-in-full-screen-modus/#post-13784540) myself also.

As an experienced WordPress user I don't use full screen mode because of this issue. It is to cumbersome to go back to wordpress and the loss of the admin-bar is not compensated. Why even remove the admin bar when in this mode? It is normally always available and should also remain imo when using the full screen mode. The advantage of the mode is more the added width than the height, and I use the toolbar too for a set of custom links that I want easy accessible and are added by my core functionality plugin.

But even when the admin toolbar remains, the problem here is that the gutenberg editor doesn't refresh on save, and so the toolbar isn't refreshed automatically. But that just means the 'view post' link should be added by Gutenberg itself, in some way, instead of relying on refresh to do it.

My own observation is (like William's) that the preview button is not a view post button, like we had before, and since I often open a document in editor and then want to (without saving) return to the frontend, it is an omission not to have such an icon on the editors toolbar. It should always be just one click away. An icon of an eye pops to mind.

Using a wordpress logo to return to the admin interface is not clear at all. When the logo is meant as a return to normal admin, than it should at least have something like a back arrow or something in it. And what corner is mostly used for stuff like an about... or a screen menu, anyway, so users are not "programmed" to look for a return-key in that place.

Why not use something like a bullet list icon, since what it returns to there is always a post-list. But anything other than the wordpress logo is needed, since that logo is used in the toolbar to show stuff like "about wordpress" and isn't intuitive at all for a return function. And when it is meant as a "return to wordpress" icon, and that is why that icon is chosen, than it implicitly communicates the editor isn't wordpress, what is obviously not what should be communicated. Now it just looks like the wordpress icon on the admin toolbar and with knowledge of the toolbar, it is expected to offer similar functionality.

The problem of preview as a way to go to the frontend view instead of having a 'view post' option is both that it takes more clicks to go to the preview than just view the document would normally take (and that means it is already a somewhat hidden feature... why not make it the default?), but also it opens a new tab and leaves the editor open (what is fine for a preview, but not for a view).

I like the idea of making the wordpress button a way to pupup the wordpress admin menu, but then a clear and intuitive and easy to reach way to exit the editor to the "normal" wordpress admin is needed. Just so a non-experienced user can easily find ways to exit it, except for using the browsers back button, I mean.

So yes, the toolbar should offer intuitive ways to add a new post, exit the editor to admin and exit the editor to view the post in the frontend. And those options should only be one mouse click away.

About the point of the "view post" pupup and it disappearance.... On a touchpad, I'm more often than not to slow to reach it in time. So either keep it for longer or make it somewhat like the chat functions we so often see nowadays, a small round icon that can open and close and perhaps even can be dragged to other places on the screen. That would offer some relieve, although having it in the outer most corner away from the publish button is also not optimal.

Probably off topic, but also a usability point is that often I find the response to publishing a message about incorrect json being returned, making it unclear if a publish is performed correctly, although often I find the post has been saved anyway. In those cases the publish button stays disabled and often I need to refresh the editor just to be able to retry the publish or save action (having to copy and paste the html to make sure I don't loose changes I made).

I understand disabling the button to prevent unneeded processing, but keeping it disabled after errors are reported, shouldn't occur.

When the above is resolved, I probably will start using full mode, since I like the idea, but for now I rather keep the admin toolbar and such available. 

Perhaps the full screen mode toggle itself should be moved out of the menu to the toolbar too (just like media players have icons for normal or full mode display) so it is easier and more intuitive to enter and exit that mode.  Having a more intuitive and always visible full mode toggle icon would also resolve some of the burden of this issue, since it would offer an easy exit of the full mode back to the admin environment.

Hope it helps. I'm a fan!

Thanks, 

Hans

**Comment by mprewitt** [comment]
Two years later and not improved yet! I just started using Gutenberg, and one day into it, this very issue is annoying the heck out of me. There should be any variation "Exit edit mode" or "View page" (either of which would do the same thing). It is super frustrating. Apparently the only option is to click the W icon, click "Return to dashboard", click "Pages" in the sidebar, and then click to "View" the page you want. Four clicks to do what was previously possible in one click. If this is representative of the blocks system overall, it is not surprising that there are so many "Disable Gutenberg this-or-that" plugins, posts, suggestions, comments.

**Comment by Kitsufoxy** [comment]
I'm another person who finds the lack of ability to exit the Gutenberg editor to be a shortcoming of the editor. Sometimes when I'm using the websites I work on with Wordpress it's just spotting a quick error that needs to be fixed and then I want to easily get back to what I was doing before I noticed the thing I needed to fix!

**Comment by youknowriad** [comment]
Closing as addressed by direction: fullscreen-by-default was a deliberate, repeatedly-reaffirmed decision but in 7.1 there's always access to the Omnibar (top bar). So I think this is mostly solved now.
